File: WoodApp-UI/src/app/models/components/customer.ts
import { AppUser } from "./appuser";

export class Customer {
    id: number = 0;
    guid: string = '';
    type: string = '';// Regular / Passenger
    firstname: string = '';
    lastname: string = '';
    fullname: string = '';
    cin: string = '';
    email: string = '';
    mfcode: string = '';
    patentecode: string = '';
    address: string = '';
    gouvernorate: string = '';
    isactive: boolean = true;
    isdeleted: boolean = false;
    maximumdiscount: number = 0;
    Maximumsalesbar: number = 0;
    notes: string = '';
    bankaccount: string = '';
    bank: string = '';
    phonenumberone: string = '';
    phonenumbertwo: string = '';
    creationdate: Date = new Date();
    updatedate: Date = new Date();
    jobtitle: string = '';
    editing: boolean = false;

    updatedbyid: number = 0;
    appuser!: AppUser;

    transporter!: Transporter;
}

export class Transporter {
    constructor(
        public id: number,
        public firstname: string,
        public lastname: string,
        public fullname: string,
        public car?: Vehicle
    ) { }
}


export class Vehicle {
    id: number = 0;
    serialnumber: string = '';
    brand!: string;
    insurancedate!: Date;
    technicalvisitdate!: Date;
    mileage!: number;
    draining!: string;
    drainingdate!: string;
    constructor(public matricule: string) {
        this.serialnumber = matricule;
    }
}

File: wood-app.api/src/ms.webapp.api.acya.core/Entities/Customer.cs
using ms.webapp.api.acya.common;
using ms.webapp.api.acya.core.Entities.CustomerDependecies;
using ms.webapp.api.acya.core.Entities.DTOs;

namespace ms.webapp.api.acya.core.Entities
{
  public class Customer : IEntity
  {
    public int Id { get; set; }
    public Guid? Guid { get; set; }
    public CounterPartType Type { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Fullname => Helpers.CapitalizeFirstLetter(FirstName!) + " " + LastName!.ToUpper();
    public string? Cin { get; set; }
    public string? Email { get; set; }
    public string? MFCode { get; set; }
    public string? PatenteCode { get; set; }
    public string? Address { get; set; }
    public string? Gouvernorate { get; set; }
    public bool? IsActive { get; set; }
    public bool? IsDeleted { get; set; }
    public double? MaximumDiscount { get; set; }
    public double? MaximumSalesBar { get; set; }
    public string? Notes { get; set; }
    public string? BankAccount { get; set; }
    public string? Bank { get; set; }
    public string? PhoneNumberOne { get; set; }
    public string? PhoneNumberTwo { get; set; }
    public DateTime? CreationDate { get; set; }
    public DateTime? UpdateDate { get; set; }
    public string? JobTitle { get; set; }

    public int UpdatedById { get; set; }
    public AppUser? AppUsers { get; set; }

    public int? TransporterId { get; set; }
    public Transporter? Transporter { get; set; }

    public Customer(){}

    public Customer(CustomerDto dto)
    {
      UpdateFromDto(dto);
    }

    public void UpdateFromDto(CustomerDto dto)
    {

    }

  }
}
